x-pack/plugins/cloud_security_posture/public/components/no_findings_states.tsx

```diff
@@ -266,24 +297,19 @@ export const NoFindingsStates = ({ postureType }: { postureType: PostureTypes })
       .filter((idxDetails) => idxDetails.status === 'unprivileged')
       .map((idxDetails: IndexDetails) => idxDetails.index)
       .sort((a, b) => a.localeCompare(b));
-  const render = () => {
-    if (status === 'not-deployed') return <NotDeployed postureType={postureType} />; // integration installed, but no agents added
-    if (status === 'indexing' || status === 'waiting_for_results') return <Indexing />; // agent added, index timeout hasn't passed since installation
-    if (status === 'index-timeout') return <IndexTimeout />; // agent added, index timeout has passed
-    if (status === 'unprivileged')
-      return <Unprivileged unprivilegedIndices={unprivilegedIndices || []} />; // user has no privileges for our indices
-    if (showConfigurationInstallPrompt)
-      return (
-        <ConfigurationFindingsInstalledEmptyPrompt
-          kspmIntegrationLink={kspmIntegrationLink}
-          cspmIntegrationLink={cspmIntegrationLink}
-        />
-      );
-  };
 
   return (
     <CloudPosturePage query={getSetupStatus}>
-      <FullSizeCenteredPage>{render()}</FullSizeCenteredPage>
+      <FullSizeCenteredPage>
+        <NoFindingsStatesNotification
+          postureType={postureType}
+          status={status}
+          unprivilegedIndices={unprivilegedIndices}
+          isNotInstalled={isNotInstalled}
+          kspmIntegrationLink={kspmIntegrationLink}
+          cspmIntegrationLink={cspmIntegrationLink}
```

Comment:
we are just forwarding these props, what about making the `EmptySecurityFindingsPrompt` get these values from the hooks?

Reply:
If what you mean is to move the status checks to the `EmptySecurityFindingsPrompt`, I would have like to, however the CloudPosturePage requires a ReactQuery object.

I did move the `kspmIntegrationLink` and `kspmIntegrationLink` hooks to the `EmptySecurityFindingsPrompt`